ci/run.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+# Builds and runs tests for a particular target passed as an argument to this
+# script.
+
+set -e
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+MANIFEST_PATH="${BASE_DIR}/Cargo.toml"
+BUILD_DIR="."
+
+VERSION="$1"
+TARGET="$2"
+
+export RUST_TEST_THREADS=1
+
+echo "======================================================="
+echo "TESTING VERSION: ${VERSION}, TARGET: ${TARGET}"
+echo "======================================================="
+
+#
+# Tell cargo what linker to use and whatever else is required
+#
+configure_cargo() {
+  rm -rf .cargo
+  mkdir .cargo
+  cp "${BASE_DIR}/ci/cargo-config" .cargo/config
+}
+
+#
+# We need to export CC for the tests to build properly (some C code is
+# compiled) to work.  We already tell Cargo about the compiler in the
+# cargo config, so we just parse that info out of the cargo config
+#
+cc_for_target() {
+  awk "/\[target\.${TARGET}\]/{getline; print}" .cargo/config |
+      cut -d '=' -f2 | \
+      tr -d '"' | tr -d ' '
+}
+
+cross_compile_tests() {
+  case "$TARGET" in
+    *-apple-ios)
+      cargo test --no-run --manifest-path="${MANIFEST_PATH}" --target "$TARGET" -- \
+            -C link-args=-mios-simulator-version-min=7.0
+      ;;
+
+    *)
+      cargo test --no-run --verbose \
+            --manifest-path="${MANIFEST_PATH}" \
+            --target "$TARGET"
+      ;;
+  esac
+}
+
+# This is a hack as we cannot currently
+# ask cargo what test files it generated:
+# https://github.com/rust-lang/cargo/issues/1924
+find_binaries() {
+  target_base_dir="${BUILD_DIR}/${TARGET}/debug"
+
+  # find [[test]] sections and print the first line and
+  # hack it to what we want from there.  Also "nix" for
+  # tests that are implicitly prsent
+  for test_base in $( awk '/\[\[test\]\]/{getline; print}' "${MANIFEST_PATH}" | \
```

Comment:
It may still be possible to collect them up by doing `$bin{-,}*` or something like that